funding/manager_test.go

```diff
@@ -3746,6 +3749,155 @@ func TestFundingManagerFundAll(t *testing.T) {
+			// We spend less than the funds in the wallet,
+			// so a change output should be created.
+			coins: constructTestUtxos(
+				2 * MaxBtcFundingAmount,
+			),
+			fundUpToMaxAmt: MaxBtcFundingAmount,
+			fundMinAmt:     MinChanFundingSize,
+			pushAmt:        0,
+			change:         true,
+		},
+		{
+			// We spend less than the funds in the wallet when
+			// setting a smaller channel size, so a change output
+			// should be created.
+			coins: constructTestUtxos(
+				MaxBtcFundingAmount,
+			),
+			fundUpToMaxAmt: MaxBtcFundingAmount / 2,
+			fundMinAmt:     MinChanFundingSize,
+			pushAmt:        0,
+			change:         true,
+		},
+		{
+			// We are using the entirety of two inputs
+			// for the funding of a channel, hence expect
+			// no change output.
+			coins: constructTestUtxos(
+				MaxBtcFundingAmount/2, MaxBtcFundingAmount/2,
+			),
+			fundUpToMaxAmt: MaxBtcFundingAmount,
+			fundMinAmt:     MinChanFundingSize,
+			pushAmt:        0,
+			change:         false,
+		},
+		{
+			// We are using a fraction of two inputs for
+			// the funding of our channel, hence expect
+			// a change output.
+			coins: constructTestUtxos(
+				MaxBtcFundingAmount/2, MaxBtcFundingAmount/2,
+			),
+			fundUpToMaxAmt: MaxBtcFundingAmount / 2,
+			fundMinAmt:     MinChanFundingSize,
+			pushAmt:        0,
+			change:         true,
+		},
+		{
+			// We are funding a channel with half of
+			// the balance in our wallet hence expect
+			// a change output. Furthermore we push
+			// half of the funding amount to the
+			// remote end which we expect to succeed.
+			coins:          constructTestUtxos(MaxBtcFundingAmount),
+			fundUpToMaxAmt: MaxBtcFundingAmount / 2,
+			fundMinAmt:     MinChanFundingSize,
+			pushAmt:        MaxBtcFundingAmount / 4,
+			change:         true,
+		},
+	}
+
+	for _, test := range tests {
+		// We set up our mock wallet to control a list of UTXOs that
+		// sum to more than the max channel size.
+		test := test
+		addFunds := func(fundingCfg *Config) {
+			wc := fundingCfg.Wallet.WalletController
+			mockWc, ok := wc.(*mock.WalletController)
+			if ok {
+				mockWc.Utxos = test.coins
+			}
+		}
+		alice, bob := setupFundingManagers(t, addFunds)
+		defer tearDownFundingManagers(t, alice, bob)
+
+		// We will consume the channel updates as we go, so no
+		// buffering is needed.
+		updateChan := make(chan *lnrpc.OpenStatusUpdate)
+
+		// Initiate a fund channel, and inspect the funding tx.
+		pushAmt := test.pushAmt
+		fundingTx := fundChannel(
+			t, alice, bob, 0, pushAmt, false, true,
+			test.fundUpToMaxAmt, test.fundMinAmt, 1, updateChan,
+			true, nil,
+		)
+
+		// Check whether the expected change output is present.
+		if test.change {
+			require.EqualValues(t, 2, len(fundingTx.TxOut))
+		}
+
+		if !test.change {
+			require.EqualValues(t, 1, len(fundingTx.TxOut))
+		}
+
+		// Inputs should be all funds in the wallet.
+		require.Equal(t, len(test.coins), len(fundingTx.TxIn))
+
+		for i, txIn := range fundingTx.TxIn {
```

Comment:
is the order of inputs here important ? Or will the funding manger rearrange funds in case we do not order them regarding to their values in the beginning?

Reply:
I think when creating the funding tx the funding manager preserves the order of the coins in the array which was passed in above... `mockWc.Utxos = test.coins`